backend/cmd/headlamp.go

```diff
@@ -1047,10 +1051,174 @@ func handleClusterHelm(c *HeadlampConfig, router *mux.Router) {
+				if cluster.Name == clusterName {
+					clusterName = cluster.Name
+					serverURL = cluster.Cluster.Server
+					if cluster.Cluster.CertificateAuthority != "" {
+						certificateAuthorityData = readFileContents(cluster.Cluster.CertificateAuthority)
+					} else {
+						certificateAuthorityData, err = base64.StdEncoding.DecodeString(cluster.Cluster.CertificateAuthorityData)
+						if err != nil {
+							log.Printf("Error: deconding certificateAuthorityData: %s", err)
+							http.NotFound(w, r)
+							return
+						}
+					}
+					clusterFound = true
+					break
+				}
+			}
+
+			for _, user := range kubeconfig.Users {
+				if user.Name == clusterName {
+					if user.User.ClientCertificate != "" {
+						clientCertificateData = readFileContents(user.User.ClientCertificate)
+					} else {
+						clientCertificateData, err = base64.StdEncoding.DecodeString(user.User.ClientCertificateData)
+						if err != nil {
+							log.Printf("Error: deconding clientCertificateData: %s", err)
+							http.NotFound(w, r)
+							return
+						}
+					}
+
+					if user.User.ClientKey != "" {
+						clientKeyData = readFileContents(user.User.ClientKey)
+					} else {
+						clientKeyData, err = base64.StdEncoding.DecodeString(user.User.ClientKeyData)
+						if err != nil {
+							log.Printf("Error: deconding clientKeyData: %s", err)
+							http.NotFound(w, r)
+							return
+						}
+					}
+					userFound = true
+					break
+				}
+			}
+
+			if !clusterFound {
+				log.Printf("Error: cluster not found")
+				http.NotFound(w, r)
+				return
+			}
+
+			if !userFound {
+				log.Printf("Error: user not found")
+				http.NotFound(w, r)
+				return
+			}
+
+			URL, err := url.Parse(serverURL)
+			if err != nil {
+				return
+			}
+
+			r.Host = URL.Host
+			r.Header.Set("X-Forwarded-Host", r.Header.Get("Host"))
+			r.URL.Host = URL.Host
+			r.URL.Path = mux.Vars(r)["api"]
+			r.URL.Scheme = URL.Scheme
+
+			if pluginsChanged() {
+				resetPlugins()
+				setPluginReloadHeader(w)
+			}
+
+			proxy := httputil.NewSingleHostReverseProxy(URL)
+			conf := api.Config{
+				Clusters: map[string]*api.Cluster{
+					clusterName: {
+						Server:                   serverURL,
+						CertificateAuthorityData: certificateAuthorityData,
+					},
+				},
+				AuthInfos: map[string]*api.AuthInfo{
+					clusterName: {
+						ClientCertificateData: clientCertificateData,
+						ClientKeyData:         clientKeyData,
+					},
+				},
+				Contexts: map[string]*api.Context{
+					clusterName: {
+						Cluster:  clusterName,
+						AuthInfo: clusterName,
+					},
+				},
+			}
+
+			clientConfig := clientcmd.NewNonInteractiveClientConfig(conf, clusterName, nil, nil)
+			restConf, err := clientConfig.ClientConfig()
+			if restConf == nil {
+				log.Printf("Error: clientConfig is nil: %s", err)
+				http.NotFound(w, r)
+				return
+			}
+
+			roundTripper, err := rest.TransportFor(restConf)
+			if err == nil {
+				proxy.Transport = roundTripper
+			}
```

Comment:
Use the `LoadContextsFromAPIConfig` from the kubeconfig package to create a Context and use the underlying `SetupProxy` function instead of repeating the logic here.

Reply:
@yolossn IIUC, I don't think we are loading context here. We are creating API client kubeconfig, so the `LoadContextsFromAPIConfig` could not be used. Am I missing something?